### DISCUSSION TITLE
Exclude own issues/pr from followup

### DISCUSSION THREAD

**Discussion by jayvdb** [lowlighter/metrics#594] [Ideas]
The followup plugin has option `plugin_followup_sections: repositories, user`.

I would like to propose a section `!user`, which would show status of issues/PRs that the owner didn't create, which can be a good gauge of the size of the users community, and how well the user maintains their project communities.

Then `plugin_followup_sections: !user, user` should be roughly the same values as `plugin_followup_sections: repositories`

**Comment by lowlighter** [comment]
Sorry I got a bit confused, but basically you're suggesting to have the `repositories` section, but excluding the issues and prs of user ?
If that's the case, then should all collaborators (i.e. push access / organizations members) be excluded from it too ?

**Reply by jayvdb** [reply]
> but excluding the issues and prs of user ?

yes

> should all collaborators (i.e. push access / organizations members) be excluded from it too ?

I had not thought about that.  Excluding anyone with write/push access would achieve what I was looking for, which is "how many issues/pr are open by people who need help" (which in my case would be a low number of 'hard' cases, that I would easily remember, and I would notice when it increases).

Another approach that would be more interesting would be to group issues/prs by access level with different colors(shades?) for each, thereby allowing a high level view of which types of issues/prs get love, and which do not.

If this does go down the path of determining project collaborators, that leads towards a very useful stats: how future proof are the users repos, as all repos used by third-parties should have someone else with access in case the bus hits.

**Reply by lowlighter** [reply]
> Another approach that would be more interesting would be to group issues/prs by access level with different colors(shades?) for each, thereby allowing a high level view of which types of issues/prs get love, and which do not.

Yeah this sounds pretty cool 👍 

> If this does go down the path of determining project collaborators, that leads towards a very useful stats: how future proof are the users repos, as all repos used by third-parties should have someone else with access in case the bus hits.

I think it's easier to ignore a list of users anyway, just so users are also able to ignore bots users too (like dependabot).
It may be an extra config though, I can see this could be more heavily API requests intensive 

Edit: Tracked in #616

**Reply by lowlighter** [reply]
So I started working on it.

I could POC the data for issues, but for pull requests it seems difficult. 
They are no filter to search by creator in both GraphQL/REST API for PRs from what I've seen, and it's obviously not possible to iterate over all pr to check creator (because of rate-limiting). Maybe could use the search endpoint, but it's not practical (because also rate limited) but may be forced to do so

**Comment by lowlighter** [comment]
Here's a preview of what it'll look like:
<img src="https://user-images.githubusercontent.com/22963968/140461268-1b5c0fd0-a793-41e5-a964-546db7b7c9e1.png" width="400">

It also includes drafts/skipped issues to match https://github.com/github/roadmap/issues/289 while drafted prs will also be added

The only drawback is that since these are not available directly through GraphQL API, it must use the Search API which has higher rate limitations
